src/instructlab/utils.py

```diff
@@ -564,3 +596,53 @@ def split_hostport(hostport: str) -> tuple[str, int]:
     if not hostname or not port:
         raise ValueError(f"Invalid host-port string: '{hostport}'")
     return hostname, port
+
+
+def convert_messages_to_legacy_dataset(
+    dataset: List[MessageSample],
+) -> List[LegacyMessageSample]:
+    """
+    Converts the new HuggingFace messages dataset format to the legacy format.
+
+    **Note**: The legacy dataset format assumes only a turn of 1. All extra turns will be dropped.
+    This means that we only look at the first 3 messages, and everything afterwards will be ignored.
+    """
+    converted_dataset: List[LegacyMessageSample] = []
+    for dp in dataset:
+        # in new dataset, the roles of a message will be determined.
+        if len(dp["messages"]) < 3:
+            raise ValueError(
+                "Dataset cannot be missing messages from roles. Please make sure that a message from each role is present."
+            )
+
+        converted: LegacyMessageSample = {  # type: ignore
+            m["role"]: m["content"] for m in dp["messages"][:3]
+        }
+        converted_dataset.append(converted)
+    return converted_dataset
+
+
+def is_messages_dataset(
+    dataset: List[MessageSample] | List[LegacyMessageSample],
+) -> bool:
+    """
+    Indicates whether or not the provided dataset is using the newer "messages" format
+    or the legacy format used by the old linux training script.
+    """
+    if len(dataset) == 0:
+        raise ValueError("dataset is empty")
+
+    return bool("messages" in dataset[0])
```

Comment:
Why do we need to cast it to `bool`? The `in` operator already guarantees that the result is `bool`. I checked if maybe `mypy` complains, - it doesn't care either.

Reply:
Good catch, your suggestion is simpler and more pythonic.